### PR TITLE
[ticket/14277] Fix undefined index error in migrations

### PR DESCRIPTION
**Pull request by CHItA**
[PHPBB3-14277](https://tracker.phpbb.com/browse/PHPBB3-14277)

### CODE REVIEW

File: phpBB/phpbb/db/migration/tool/module.php [hunk 1]
```diff
@@ -262,7 +262,7 @@ public function add($class, $parent = 0, $data = array())
 
 			// Success
 			$module_log_name = ((isset($this->user->lang[$data['module_langname']])) ? $this->user->lang[$data['module_langname']] : $data['module_langname']);
-			$phpbb_log->add('admin', $user->data['user_id'], $user->ip, 'LOG_MODULE_ADD', false, array($module_log_name));
+			$phpbb_log->add('admin', ANONYMOUS, $user->ip, 'LOG_MODULE_ADD', false, array($module_log_name));
```
Comment (marc1706) [inline phpBB/phpbb/db/migration/tool/module.php]: I'd prefer checking if user id exists and use it if it is. Or will this always be undefined?

Comment (CHItA) [inline phpBB/phpbb/db/migration/tool/module.php]: As far as I'm aware yes. But that might change in the future.

File: phpBB/assets/javascript/installer.js [hunk 2]
```diff
@@ -243,8 +243,21 @@
...
 	function parseMessage(messageJSON) {
 		$('#loading_indicator').css('display', 'none');
+		var responseObject;
 
-		var responseObject = JSON.parse(messageJSON);
+		try {
+			responseObject = JSON.parse(messageJSON);
+		}
+		catch (err) {
+			if (window.console) {
```
Comment (marc1706) [inline phpBB/assets/javascript/installer.js]: I'd rather not output the error via console or alert. Just catch it and reset the polling by default.

File: phpBB/assets/javascript/installer.js [hunk 3]
```diff
@@ -243,8 +243,21 @@
 	 */
 	function parseMessage(messageJSON) {
 		$('#loading_indicator').css('display', 'none');
+		var responseObject;
 
-		var responseObject = JSON.parse(messageJSON);
+		try {
+			responseObject = JSON.parse(messageJSON);
+		}
+		catch (err) {
```
Comment (marc1706) [inline phpBB/assets/javascript/installer.js]: Needs to be ```} catch (err) {```

File: phpBB/assets/javascript/installer.js (hunk 2 above)
Comment (CHItA) [inline phpBB/assets/javascript/installer.js]: I disagree, as in that case, you get no notification about the installer failing.

Comment (marc1706) [inline phpBB/assets/javascript/installer.js]: I guess we can leave this in for now but I'd rather not have alert messages popping up in a stable release ...

Comment (CHItA) [inline phpBB/assets/javascript/installer.js]: only in IE8

Comment (marc1706) [inline phpBB/assets/javascript/installer.js]: Anyway, pls poke me once tests finished so I can try again ;)

**Pull request merged**